posts/2019-09-18-upcoming-docsrs-changes.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Upcoming docs.rs changes"
+author: The Rust Infrastructure Team
+---
+
+On September 42nd breaking changes will be deployed to the [docs.rs] build
+environment. [docs.rs] is a free service building and hosting documentation for
+all the crates published on [crates.io]. It's [open source][docsrs-source] and
+maintained by the [Rustdoc team][rustdoc-team].
+
+## What will change
+
+Builds will be executed inside the [rustops/crates-build-env] Docker image.
+That image contains a lot of system dependencies installed to ensure we can
+build as many crates as possible. It's already used by [Crater], and we added
+all the dependencies previously installed in the legacy build environment.
+
+To ensure we can continue operating the service in the future and to increase
+its reliability we also improved the sandbox builds are executed into, adding
+new limits:
+
+* Each platform will now have **15 minutes** to build its dependencies and
+  documentation.
+* **3 GB of RAM** will be available for the build.
+* Network access will be completly **disabled**.
+* Only the `target/` directory will be writable, and it will be purged after
+  each build.
+
+Finally, docs.rs will now use the latest nightly available when building
+crates.
+
+## How to prepare for the changes
+
+To test if your crate builds inside the new environment you can download the
+Docker image locally and execute a shell inside it:
+
+```
+docker pull rustops/crates-build-env
+docker run --rm -it rustops/crates-build-env bash
+```
+
+Once you're in a shell you can install [rustup] (it's not installed in the
+image), install Rust nightly, clone your crate's repository and then build the
+documentation:
+
+```
+cargo doc --no-deps
+```
+
+If a dependency is missing please [open an issue][crates-build-env-issue] on
+the Docker image's repository.
+
+If your crate fails to build because it took more than 15 minutes to generate
+its docs or it uses more than 3 GB of RAM please [open an issue][docsrs-issue]
+and we will increase the limits for your crate (when it's reasonable). We will
+**not** enable network access for your crate though: you'll need to change your
+crate not to require any external resource at build time.
```

Comment:
Would it be be useful to show how this might work? Are there any well-known crates where we could link to a git commit where they did this?

Reply:
Not sure I have examples myself. Some FFI bindings like rusqlite bundle their C libraries if you pass a feature flag, but that's not really the same thing?